architecture/2025-04-aci-confidential-containers.md

```diff
@@ -0,0 +1,388 @@
+6. Create the container group profile using the ACI SDK with the CCE policy set.
+
+#### Advantages (of each option considered)
+<!--
+Describe what's good about this plan relative to other options. 
+Provides better user experience? Does it feel easy to implement? 
+Provides flexibility for future work?
+-->
+
+This approach ensures the user experience has parity with Kubernetes `runtimes`. Radius takes care of setting up the backing resources for a container instance, including generating the necessary security policies.
+
+#### Disadvantages (of each option considered)
+<!--
+Describe what's not ideal about this plan. Does it lock us into a 
+particular design for future changes or is it flexible if we were to 
+pivot in the future. This is a good place to cover risks.
+-->
+
+The CCE policy can only be generated using the `confcon` CLI extension. There are no available SDKs to generate this value. As a result, Radius must have access to the extension to generate policies on behalf of the user. 
+
+### API design (if applicable)
+
+<!--
+Include if applicable – any design that changes our public REST API, CLI
+arguments/commands, or Go APIs for shared components should provide this
+section. Write N/A here if not applicable.
+- Describe the REST APIs in detail for new resource types or updates to
+  existing resource types. E.g. API Path and Sample request and response.
+- Describe new commands in the CLI or changes to existing CLI commands.
+- Describe the new or modified Go APIs for any shared components.
+-->
+
+`typespec/Applications.Core/containers.tsp`
+
+```typespec
+@doc("The properties for runtime configuration")
+model RuntimesProperties {
+  @doc("The runtime configuration properties for Kubernetes")
+  kubernetes?: KubernetesRuntimeProperties;
+
+  @doc("The runtime configuration properties for ACI")
+  aci?: ACIRuntimeProperties;
+}
+
+#suppress "@azure-tools/typespec-azure-core/bad-record-type"
+@doc("A merge will be applied to the ContainerGroupProfile object when this container is being deployed.")
+model ContainerGroupProfile is Record<unknown>;
+
+@doc("The runtime configuration properties for ACI")
+model ACIRuntimeProperties {
+  #suppress "@azure-tools/typespec-azure-core/bad-record-type"
+  @doc("A merge will be applied to the ContainerGroupProfile object when this container is being deployed.")
+  containergroupprofile?: ContainerGroupProfile;
+}
+```
+
+### CLI Design (if applicable)
+<!--
+Include if applicable – any design that changes Radius CLI
+arguments/commands. Write N/A here if not applicable.
+- Describe new commands in the CLI or changes to existing CLI commands.
+-->
+
+`rad init` would add 2 steps: one step to determine which compute platform to use when the user chooses to set up an environment and one step to choose confidential containers. See the user experience section for details. 
+
+### Error Handling
```

Comment:
How/where do we handle issues with issues with installation of extension or missing az cli?

Reply:
The issue will be surfaced in the container group profile handler since that's where we'd use the extension. We'd require the extension as a prerequisite so the failure will let the user know they need to install it. I'll update the doc with that info